test/shared/src/main/scala/zio/test/TestRunner.scala

```diff
@@ -29,12 +29,14 @@ import zio.internal.Platform
 final case class TestRunner[R <: Has[_], E](
   executor: TestExecutor[R, E],
   platform: Platform = Platform.makeDefault().withReportFailure(_ => ()),
-  reporter: TestReporter[E] = DefaultTestReporter(TestAnnotationRenderer.default),
```

Comment:
It seems like we are losing the ability to supply different test annotation renderers here right now. Maybe you pass both a test renderer and a test annotation renderer? Or maybe you pass a test reporter and you can easily construct a test reporter from a test renderer and an annotation renderer?